src/hash_builder/mod.rs

```diff
@@ -201,20 +216,27 @@ impl HashBuilder {
         let mut build_extensions = false;
         // current / self.key is always the latest added element in the trie
         let mut current = self.key.clone();
+
         debug_assert!(!current.is_empty());
 
         trace!(target: "trie::hash_builder", ?current, ?succeeding, "updating merkle tree");
 
         let mut i = 0usize;
+        let trie_default = &TrieMask::default();
```

Comment:
a nicer approach here would be to introduce a `TrieMask::empty()` const fn and use it in all places instead of introducing a variable